cmd/main.go

```diff
@@ -98,15 +97,18 @@ func Main(args []string) error {
 			args = []string{"mount", "--help"}
 		}
 	}
-	err := app.Run(reorderOptions(app, args))
+	options := reorderOptions(app, args)
+	// replace original args with handled args to pass to child process
+	os.Args = options
+	err := app.Run(options)
 	if errno, ok := err.(syscall.Errno); ok && errno == 0 {
 		err = nil
 	}
 	return err
 }
 
 func calledViaMount(args []string) bool {
-	return strings.HasSuffix(args[0], "/mount.juicefs")
+	return strings.HasSuffix(args[0], "/mount.juicefs") && args[1] != "mount"
```

Comment:
What if the volume is called `mount`?

Reply:
`handleSysMountArgs`  cannot handle correctly `/mount.juicefs mount -d redis://127.0.0.1 /tmp/jfs`
We previously thought that `mount.juicefs` should be used without the `mount` subcommand.